src/shared/GlobalBanners/SelfHostedLicenseExpiration/SelfHostedLicenseExpiration.tsx

```diff
@@ -118,4 +116,18 @@ const SelfHostedLicenseExpiration = () => {
   )
 }
 
-export default SelfHostedLicenseExpiration
+// This wrapper is just so we don't make a request to the API if we're not on a
+// self-hosted instance, this is because we're useSuspenseQuery is not
+// toggle'ble through a `enabled` field like useQuery
+function SelfHostedLicenseExpirationWrapper() {
+  const isSelfHosted = !!config.IS_SELF_HOSTED
+  const isDedicatedNamespace = !!config.IS_DEDICATED_NAMESPACE
+
+  if (!isSelfHosted || !isDedicatedNamespace) {
+    return null
+  }
+
+  return <SelfHostedLicenseExpiration />
+}
+
+export default SelfHostedLicenseExpirationWrapper
```

Comment:
I am probably missing it - where is this wrapper component meant to be used?

Was just wondering in case perhaps there's a way to create a generic "conditionalSuspense" or similar wrapper to handle if we have this scenario in other places

Reply:
So i played around a bit with this idea and came up with something like:

```tsx
// ConditionalRenderer.tsx
export const ConditionalRenderer = ({ condition, children }) => {
  if (condition) {
    return <>{children}</>
  }

  return null
}

// SelfHostedLicenseExpiration.tsx
function SelfHostedLicenseExpirationWrapper() {
  const isSelfHosted = !!config.IS_SELF_HOSTED
  const isDedicatedNamespace = !!config.IS_DEDICATED_NAMESPACE

  return (
    <ConditionalRenderer condition={isSelfHosted && isDedicatedNamespace}>
      <SelfHostedLicenseExpiration />
    </ConditionalRenderer>
  )
}

export default SelfHostedLicenseExpirationWrapper
```

Because of the way TanStack suspense queries work, we're still required to creating this wrapping component to conditionally render the hook call, so the resulting abstraction becomes fairly simple. Because of this I'd rather avoid trying to abstract it, and just handle the conditions directly in the same file. 

The main motivating factor behind keeping the conditional rendering in the same file, is that everything is defined up front right in front of the engineer, and if there are more complex condition cases we're not stuck trying to create a "one size fits all" conditional rendering component that is a pain to maintain and makes the app overall more fragile.